<a href="https://colab.research.google.com/github/EricAndradeNJIT/data-analytics-folder/blob/main/Heigh_Prediction_Linear_R_One_hot.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files
heights = files.upload()

Saving heights.csv to heights.csv


In [2]:
#import packages

import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_squared_error as MSE

In [3]:
heights = pd.read_csv('heights.csv')

In [4]:
#quick glance data

print(heights.shape)

heights.head()

(898, 7)


Family  Father  Mother Gender  Height  Kids  MidParent
0       1    78.5    67.0      M    73.2     4      75.43
1       1    78.5    67.0      F    69.2     4      75.43
2       1    78.5    67.0      F    69.0     4      75.43
3       1    78.5    67.0      F    69.0     4      75.43
4       2    75.5    66.5      M    73.5     4      73.66

In [7]:
X = heights[['MidParent', 'Gender', 'Family']]

y = heights[['Height']]

print(y)
print(X)

     Height
0      73.2
1      69.2
2      69.0
3      69.0
4      73.5
..      ...
893    68.5
894    67.7
895    64.0
896    63.5
897    63.0

[898 rows x 1 columns]
     MidParent Gender  Family
0        75.43      M       1
1        75.43      F       1
2        75.43      F       1
3        75.43      F       1
4        73.66      M       2
..         ...    ...     ...
893      69.35      M     205
894      69.35      M     205
895      69.35      F     205
896      69.35      F     205
897      69.35      F     205

[898 rows x 3 columns]


In [9]:
#one code features of interest

one_hot = pd.get_dummies(data=X, columns=['Gender', 'Family'])

print(one_hot.head())

   MidParent  Gender_F  Gender_M  Family_1  Family_2  Family_3  Family_4  \
0      75.43         0         1         1         0         0         0   
1      75.43         1         0         1         0         0         0   
2      75.43         1         0         1         0         0         0   
3      75.43         1         0         1         0         0         0   
4      73.66         0         1         0         1         0         0   

   Family_5  Family_6  Family_7  ...  Family_195  Family_196  Family_197  \
0         0         0         0  ...           0           0           0   
1         0         0         0  ...           0           0           0   
2         0         0         0  ...           0           0           0   
3         0         0         0  ...           0           0           0   
4         0         0         0  ...           0           0           0   

   Family_198  Family_199  Family_200  Family_201  Family_203  Family_204  \
0        

In [10]:
print(one_hot.columns)

Index(['MidParent', 'Gender_F', 'Gender_M', 'Family_1', 'Family_2', 'Family_3',
       'Family_4', 'Family_5', 'Family_6', 'Family_7',
       ...
       'Family_195', 'Family_196', 'Family_197', 'Family_198', 'Family_199',
       'Family_200', 'Family_201', 'Family_203', 'Family_204', 'Family_205'],
      dtype='object', length=200)


In [13]:
X = pd.DataFrame(one_hot)



In [25]:
#split data into training/validation and test datasets

X_train_val, X_test, y_train_val, y_test = train_test_split(X, y, test_size=0.25, random_state=42)
print(y_train_val)

     Height
377    63.0
357    64.0
723    71.5
306    64.5
464    66.5
..      ...
106    66.7
270    72.0
860    71.0
435    65.0
102    66.5

[673 rows x 1 columns]


In [26]:
#split validation dataset into val and train setsd

X_train, X_val, y_train, y_val = train_test_split(X_train_val, y_train_val, test_size=0.3333, random_state=42) 


In [27]:
 print(X_val.shape)
 X_train.shape

(225, 200)


(448, 200)

In [31]:
#fit linear regression

reg = LinearRegression()

model = reg.fit(X_train, y_train)

In [34]:
#calculate predicted values

y_pred = model.predict(X_train)

y_pred.shape

(448, 1)

In [35]:
#interpret RMSE and R-squared using training data

RMSE_train = MSE(y_train, y_pred, squared=False)

print(RMSE_train)

1.5723719090945663


In [36]:
r2_train = r2_score(y_train, y_pred)

print(r2_train)

0.8115693166694996


The average difference between an observed child's height and the height predicted is 1.6 inches

About 81% of the variability in the height of adult children can be explained by the height of the parents, their gender and the family of origin.